### DISCUSSION TITLE
Consider add a retrived documents section for reference?

### DISCUSSION THREAD

**Discussion by NeverOccurs** [jonfairbanks/local-rag#31] [Ideas]
Really love your work. This may be the best RAG application use so far that I've ever tried, easy to install and customizable LLM and embeddings. Great for personal use. One piece of advice would be adding a feature that shows relevant chunks or documents that the LLM generated responses from with scores ranking from high to low, like Danswer or CheshireCat? (Personally I prefer the way Danswer is doing) It will be a great feature for academic or research use. Thanks in advance!

**Comment by jonfairbanks** [comment]
Thank you for the suggestion! I have not used Danswer yet but will give it a shot and see if I can implement something similar over here. In the meantime if anyone else has a chance to implement this type of feature, would happily accept a PR!
